dj/sql/parsing/ast.py

```diff
@@ -1030,6 +1030,20 @@ def __str__(self) -> str:
         return prefix + str(self.name)
 
 
+@dataclass(eq=False)
```

Comment:
I think I spoke too soon about my option "c" in your issue. This looks good. Are there any other cases like this where there's some special syntax on a column that would parse to something other than `MapSubscript`?

Reply:
The only type of map subscription syntax I can think of is `some_map["some_key"]`, but I suppose every dialect has their own version of this type of map access. In fact, I'm not even sure that `some_map["some_key"]` is ANSI SQL, but it turns out sqloxide supports it!